src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentCopy = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      currentCopy = {};
+    } else if (action.type === 'addProperties') {
+      currentCopy = { ...currentCopy, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      currentCopy = { ...currentCopy };
```

Comment:
Creating a new copy of `currentCopy` on line 19 is unnecessary since you are not making any changes to the object structure that would require a new copy. You can remove this line.